.github/workflows/CI.yml

```diff
@@ -92,38 +92,41 @@ jobs:
       - name: Execute tests
         run: cd build && ctest --output-on-failure
 
-  build-pip:
-    runs-on: ubuntu-latest
-    container: mrtravis/gridtools:${{ matrix.compiler }}
+  build-python:
     strategy:
       matrix:
-        compiler: [gcc-7, gcc-8, gcc-9, clang-9]
-        build_type: [Debug, Release]
-        exclude:
-          - compiler: gcc-7
-            build_type: Debug
-          - compiler: gcc-8
-            build_type: Debug
-          - compiler: clang-9
-            build_type: Debug
-
+        os: [ubuntu-latest, macos-latest]
+        python-version: [ "3.10"]
+    runs-on: ${{ matrix.os }}
     steps:
-      - uses: actions/checkout@v1
-      - name: Set defaults
-        run: |
-          export GCC_VERSION=$(echo ${{ matrix.compiler }} | cut -d'-' -f2)
-          echo "FC=gfortran-${GCC_VERSION}" >> $GITHUB_ENV
-          echo "CMAKE_ARGS=-DCMAKE_BUILD_TYPE=${{ matrix.build_type }} -DCMAKE_C_COMPILER_LAUNCHER=ccache -DCMAKE_CXX_COMPILER_LAUNCHER=ccache" >> $GITHUB_ENV
-          echo "TEST_FORTRAN=OFF" >> $GITHUB_ENV  # default will be overwritten in the next step
-      - name: Install python dependencies
+      - uses: actions/checkout@v3
+      - name: install boost on macos
+        if: ${{ matrix.os  == 'macos-latest'}}
         run: |
-          pip3 --version
-          pip3 install --upgrade pip setuptools wheel
-          pip3 install --user nose numpy
-      - name: Configure
-        run: ccache -s
-      - name: Build
+            brew install boost
+      - name:  install boost for ubuntu
+        if: ${{ matrix.os   == 'ubuntu-latest'}}
         run: |
-          pip3 install src/serialbox-python
-      - name: Execute tests
-        run: python3 -m nose test/serialbox-python/serialbox
+            sudo apt-get update
+            sudo apt-get install libboost-all-dev
+
+      - name: Set up Python ${{ matrix.python-version }}
+        uses: actions/setup-python@v4
+        with:
+            python-version: ${{ matrix.python-version }}
+            cache: 'pip'
+
+      - name:  upgrade python tools
+        run: python -m pip install --upgrade pip setuptools wheel
+
+      - name: install dependencies
+        run: pip install -r src/serialbox-python/requirements.txt
+
+      - name: build
+        run:  pip install src/serialbox-python
+
+      - name: run tests
+        run: pytest -v test/serialbox-python/serialbox >> $GITHUB_STEP_SUMMARY
+
+      - name: clean up
+        run: pip uninstall --yes serialbox
```

Comment:
Why? Is this explicitly testing that uninstall doesn't break?

Reply:
Don't know whether this makes sense, I added it because of the caching. I don't want the serialbox installation to be cached.

Reply:
There should be no caching. Maybe let's remove then, as later we might be confused why it's there.